src/main/java/service/handler/SupplyHandler.java

```diff
@@ -0,0 +1,14 @@
+package service.handler;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public class SupplyHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction,
+                      Map<String, Integer> fruitBalance) {
+        fruitBalance.put(fruitTransaction.getFruit(),
+                fruitBalance.get(fruitTransaction.getFruit())
+                        + fruitTransaction.getQuantity());
+    }
```

Comment:
instead of this you can use map's merge method, with Integer::sum in this handler